node_modules/oae-meetups/lib/api.meetups.js

```diff
@@ -0,0 +1,294 @@
+
+            _setMeetupLayout(doc);
+
+            var xml = serializer.serializeToString(doc);
+            MeetupsAPI.Bbb.joinURL(ctx, profile, xml, function(err, joinInfo) {
+                if(err) {
+                    //res.send(503, 'Fatal error'); *res is not defined here...*
+                    log().info('Fatal error');
+                }
+
+                MeetupsAPI.emit(MeetupsConstants.events.JOIN_MEETUP, ctx, groupProfile, function() {
+                    //handle errs
+                });
+
+                return callback(null, joinInfo);
+            });
+
+            return undefined;
+        });
+
+        return undefined;
+    });
+
+    return undefined;
+};
+
+var _setMeetupLayout = function(doc) {
+    var select = xpath.useNamespaces();
+    var node;
+
+    // set layout bbb.layout.name.videochat and others
+    node = select('//layout ', doc, true);
+    node.setAttribute('defaultLayout', 'bbb.layout.name.videochat');
+    node.setAttribute('showLayoutTools', 'false');
+    node.setAttribute('confirmLogout', 'false');
+    node.setAttribute('showRecordingNotification', 'false');
+    // process modules
+    // remove desktop sharing
+    node = xpath.select1("//modules/module[@name=\'ScreenshareModule\']", doc);
+    node.setAttribute('showButton', 'false');
+    // remove PhoneModule button
+    node = xpath.select1("//modules/module[@name=\'PhoneModule\']", doc);
+    node.setAttribute('showButton', 'true');
+    node.setAttribute('skipCheck', 'true');
+    node.setAttribute('listenOnlyMode', 'false');
+    // remove VideoconfModule button
+    node = xpath.select1("//modules/module[@name=\'VideoconfModule\']", doc);
+    node.setAttribute('showButton', 'true');
+    node.setAttribute('autoStart', 'true');
+    node.setAttribute('skipCamSettingsCheck', 'true');
+    // remove layout menu
+    node = xpath.select1("//modules/module[@name=\'LayoutModule\']", doc);
+    node.setAttribute('enableEdit', 'false');
+}
+
+/**
+ *
+ * Checks if a meeting is running
+ *
+ * @function isMeetingRunning
+ * @param  {type} ctx      {description}
+ * @param  {type} groupId  {description}
+ * @param  {type} callback {description}
+ * @return {type} {description}
+ */
+var isMeetingRunning = function (ctx, groupId, callback) {
+    var validator = new Validator();
+    validator.check(null, {'code': 401, 'msg': 'Only authenticated users can end meetups'}).isLoggedInUser(ctx);
+    validator.check(groupId, {'code': 400, 'msg': 'Invalid groupId id provided'}).isResourceId();
+
+    PrincipalsAPI.getFullGroupProfile(ctx, groupId, function(err, groupProfile) {
+        if(err) {
+            callback(err);
+        }
+
+        // TODO this is not quite right, this could come from the activity stream
+        // outside of a group, in which case this ajax call shouldn't be happening
+        var profile;
+        try {
+            profile = (JSON.parse(JSON.stringify(groupProfile)));
```

Comment:
I wasn't the one who wrote this, but I searched around online and it might be to validate the JSON. Thoughts?